### ISSUE TITLE
String template plugin in Rider (2022.3.1) fails to open template file (infinite spinning wheel)

### ISSUE THREAD

**Issue by LennyThompson** [antlr/jetbrains-plugin-st4#41]
Rider 2022.3.1 with Stringtemplate 0.9 and antlr 1.9.3-3 plugins installed.

Opening any string template file never succeeds, and just have an infinite (for my level of patience anyway) spinning wheel.

Using Intellij the same template files are opened with correct syntax highlighting etc.

The following error gets reported in the Rider idea.log after opening a string template file.

```
2023-01-11 11:07:02,614 [  62582] SEVERE - #o.j.c.AsyncPromise - Cannot invoke "java.util.List.get(int)" because "org.antlr.jetbrains.st4plugin.psi.STGroupTokenTypes.RULE_ELEMENT_TYPES" is null

java.lang.NullPointerException: Cannot invoke "java.util.List.get(int)" because "org.antlr.jetbrains.st4plugin.psi.STGroupTokenTypes.RULE_ELEMENT_TYPES" is null

at org.antlr.jetbrains.st4plugin.psi.STGroupTokenTypes.getRuleElementType(STGroupTokenTypes.java:19)

at org.antlr.jetbrains.st4plugin.psi.STGroupParserDefinition.createElement(STGroupParserDefinition.java:81)

at com.intellij.psi.impl.source.tree.CompositeElement.createPsiNoLock(CompositeElement.java:705)

at com.intellij.psi.impl.source.tree.CompositeElement.getPsi(CompositeElement.java:692)

at com.intellij.psi.impl.source.SourceTreeToPsiMap.treeElementToPsi(SourceTreeToPsiMap.java:16)

at com.intellij.psi.impl.source.tree.SharedImplUtil.getParent(SharedImplUtil.java:33)

at com.intellij.psi.impl.source.tree.LeafPsiElement.getParent(LeafPsiElement.java:70)

at com.intellij.completion.ml.common.RecentPlacesFeatures$StoreRecentPlacesListener.findDeclaration(RecentPlacesFeatures.kt:104)

at com.intellij.completion.ml.common.RecentPlacesFeatures$StoreRecentPlacesListener.recentPlaceAdded$lambda$1(RecentPlacesFeatures.kt:73)

at com.intellij.util.RunnableCallable.call(RunnableCallable.java:20)

at com.intellij.util.RunnableCallable.call(RunnableCallable.java:11)

at com.intellij.openapi.application.impl.NonBlockingReadActionImpl$OTelMonitor.callWrapped(NonBlockingReadActionImpl.java:746)

at com.intellij.openapi.application.impl.NonBlockingReadActionImpl$OTelMonitor$MonitoredComputation.call(NonBlockingReadActionImpl.java:778)

at com.intellij.openapi.application.impl.NonBlockingReadActionImpl$Submission.insideReadAction(NonBlockingReadActionImpl.java:573)

at com.intellij.openapi.application.impl.NonBlockingReadActionImpl$Submission.lambda$attemptComputation$3(NonBlockingReadActionImpl.java:537)

at com.intellij.openapi.application.impl.ApplicationImpl.tryRunReadAction(ApplicationImpl.java:1086)

at com.intellij.openapi.progress.util.ProgressIndicatorUtils.lambda$runInReadActionWithWriteActionPriority$0(ProgressIndicatorUtils.java:71)

at com.intellij.openapi.progress.util.ProgressIndicatorUtilService.runActionAndCancelBeforeWrite(ProgressIndicatorUtilService.java:63)

at com.intellij.openapi.progress.util.ProgressIndicatorUtils.runActionAndCancelBeforeWrite(ProgressIndicatorUtils.java:128)

at com.intellij.openapi.progress.util.ProgressIndicatorUtils.lambda$runWithWriteActionPriority$1(ProgressIndicatorUtils.java:109)

at com.intellij.openapi.progress.ProgressManager.lambda$runProcess$0(ProgressManager.java:68)

at com.intellij.openapi.progress.impl.CoreProgressManager.lambda$runProcess$2(CoreProgressManager.java:188)

at com.intellij.openapi.progress.impl.CoreProgressManager.lambda$executeProcessUnderProgress$13(CoreProgressManager.java:589)

at com.intellij.openapi.progress.impl.CoreProgressManager.registerIndicatorAndRun(CoreProgressManager.java:664)

at com.intellij.openapi.progress.impl.CoreProgressManager.computeUnderProgress(CoreProgressManager.java:620)

at com.intellij.openapi.progress.impl.CoreProgressManager.executeProcessUnderProgress(CoreProgressManager.java:588)

at com.intellij.openapi.progress.impl.ProgressManagerImpl.executeProcessUnderProgress(ProgressManagerImpl.java:60)

at com.intellij.openapi.progress.impl.CoreProgressManager.runProcess(CoreProgressManager.java:175)

at com.intellij.openapi.progress.ProgressManager.runProcess(ProgressManager.java:68)

at com.intellij.openapi.progress.util.ProgressIndicatorUtils.runWithWriteActionPriority(ProgressIndicatorUtils.java:106)

at com.intellij.openapi.progress.util.ProgressIndicatorUtils.runInReadActionWithWriteActionPriority(ProgressIndicatorUtils.java:71)

at com.intellij.openapi.application.impl.NonBlockingReadActionImpl$Submission.attemptComputation(NonBlockingReadActionImpl.java:537)

at com.intellij.openapi.application.impl.NonBlockingReadActionImpl$Submission.lambda$transferToBgThread$1(NonBlockingReadActionImpl.java:444)

at com.intellij.codeWithMe.ClientId$Companion.decorateRunnable$lambda$6(ClientId.kt:308)

at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)

at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)

at java.base/java.util.concurrent.Executors$PrivilegedThreadFactory$1$1.run(Executors.java:702)

at java.base/java.util.concurrent.Executors$PrivilegedThreadFactory$1$1.run(Executors.java:699)

at java.base/java.security.AccessController.doPrivileged(AccessController.java:399)

at java.base/java.util.concurrent.Executors$PrivilegedThreadFactory$1.run(Executors.java:699)

at java.base/java.lang.Thread.run(Thread.java:833)

```

**Comment by parrt** [comment]
That's very strange. Unfortunately I haven't used that plug-in in years and I'm stuck in a bunch of other open source projects at the moment. I'm not sure why rebooting causes that problem.

**Comment by BillBaird** [comment]
I can confirm that this problem exists in Rider 2022.3.2 as well.  I first noticed it after installing 2022.3.1, and have kept it disabled ever since.  While not critical, I do miss the syntax highlighting.

**Comment by bjansen** [comment]
Fixed by #44
